### PR TITLE
Course templates

### PR DESCRIPTION
**Pull request by joshcosta**
This is ready for initial review! I'll be adding tests in the meantime. To test it, also pull the course-templates branch in client and follow the UI cues, watching network activity.

- Adds the courses/import endpoint for course template sharing
- DRYs creating "full" courses and refactors demo course creation to use this

### CODE REVIEW

**Comment by joshcosta** [comment]
We need to decide what to do about lint complaining about awaits in for loops. As far as I can tell they're pretty necessary here

**Comment by ramsays** [comment]
I guess normally we would use `Promise.all` for that, but I guess the way we have it implemented here we should just tell lint to ignore that line. Thoughts?

**Comment by joshcosta** [comment]
I'll defer to @vikaspotluri123 on that one

File: test/fixtures/example-data.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const fixtures = [
 	['courses', {id: '5d3c822b361ac86457c4395e', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'Course C', cut1: 90, cut1Name: 'A', cut2: 80, cut2Name: 'B', cut3: 70, cut3Name: 'C', cut4: 60, cut4Name: 'D', credit_hours: null}],
 	['courses', {id: '5d3c8230c95745a3e0806c9b', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'Course D', cut1: 90, cut1Name: 'A', cut2: 80, cut2Name: 'B', cut3: 70, cut3Name: 'C', cut4: 60, cut4Name: 'D', credit_hours: null}],
 	['courses', {id: '5d48437d8c01946f39ea997f', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'Course No Grades', cut1: 90, cut1Name: 'A', cut2: 80, cut2Name: 'B', cut3: 70, cut3Name: 'C', cut4: 60, cut4Name: 'D', credit_hours: null}],
+	['imports', {semester: '2019S', name: 'Course to imports', cut1: 90, cut1Name: 'A', cut2: 80, cut2Name: 'B', cut3: 70, cut3Name: 'C', cut4: 60, cut4Name: 'D', credits: null}],
```
Comment (vikaspotluri123) [inline test/fixtures/example-data.js:12]: I think this breaks the init script; Can you add a filter that removes this in that script?

**Comment by joshcosta** [comment]
I think I'm about finished here. Let me know if you see anything else that needs work

**Pull request merged**